arch/arm/src/arm/arm_dataabort.c

```diff
@@ -70,15 +70,16 @@ void arm_dataabort(uint32_t *regs, uint32_t far, uint32_t fsr)
 {
   struct tcb_s *tcb = this_task();
 #ifdef CONFIG_LEGACY_PAGING
-  uint32_t *savestate;
+  bool savestate;
 
   /* Save the saved processor context in current_regs where it can be
    * accessed for register dumps and possibly context switching.
    */
 
-  savestate = up_current_regs();
+  savestate = arm_get_irq_flag();
 #endif
-  up_set_current_regs(regs);
+  tcb->xcp.regs = regs;
```

Comment:
This will now destroy the saved task context in this case:
1) Task is running in thread mode
2) Task starts serving an interrupt -> save task context to tcb->xcp.regs
3) Task gets a data abort exception in the kernel -> save task context to tcb->xcp.regs overriding the context save in 2)

Reply:
Can't a page fault exception occur in the kernel when e.g. trying to access a user page that has been swapped? This would invoke the page fault exception when running the interrupt handler in kernel.

I don't know if this can happen right now in practice, but it is theoretically possible.